### PR TITLE
r/sagemaker_notebook_instance - `lifecycle_config_name`, `root_access`, and `default_code_repository` allow updating + refactor tests

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #8880

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_sagemaker_notebook_instance:  `lifecycle_config_name` and `root_access`  are updateable.
resource_aws_sagemaker_notebook_instance: plan time validation for `role_arn`, `instance_type`.
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_'
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (353.11s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (728.56s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (721.73s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (360.70s)
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (341.23s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (648.05s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (428.07s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (811.45s)
```

### CODE REVIEW

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_ -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_update
=== PAUSE TestAccAWSSagemakerNotebookInstance_update
=== RUN   TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== PAUSE TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== RUN   TestAccAWSSagemakerNotebookInstance_tags
=== PAUSE TestAccAWSSagemakerNotebookInstance_tags
=== RUN   TestAccAWSSagemakerNotebookInstance_kms
=== PAUSE TestAccAWSSagemakerNotebookInstance_kms
=== RUN   TestAccAWSSagemakerNotebookInstance_disappears
=== PAUSE TestAccAWSSagemakerNotebookInstance_disappears
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== RUN   TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
=== CONT  TestAccAWSSagemakerNotebookInstance_kms
=== CONT  TestAccAWSSagemakerNotebookInstance_tags
=== CONT  TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== CONT  TestAccAWSSagemakerNotebookInstance_update
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (375.20s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (404.47s)
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
    resource_aws_sagemaker_notebook_instance_test.go:255: [INFO] Got non-empty plan, as expected
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (465.30s)
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (540.23s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (677.81s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (686.32s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (715.37s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (738.39s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	738.437s
```

**Review by ewbankkit — COMMENTED** [review]
Some acceptance test failures:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_ -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_update
=== PAUSE TestAccAWSSagemakerNotebookInstance_update
=== RUN   TestAccAWSSagemakerNotebookInstance_volumesize
=== PAUSE TestAccAWSSagemakerNotebookInstance_volumesize
=== RUN   TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== PAUSE TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== RUN   TestAccAWSSagemakerNotebookInstance_tags
=== PAUSE TestAccAWSSagemakerNotebookInstance_tags
=== RUN   TestAccAWSSagemakerNotebookInstance_kms
=== PAUSE TestAccAWSSagemakerNotebookInstance_kms
=== RUN   TestAccAWSSagemakerNotebookInstance_disappears
=== PAUSE TestAccAWSSagemakerNotebookInstance_disappears
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== RUN   TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== RUN   TestAccAWSSagemakerNotebookInstance_default_code_repository
=== PAUSE TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
=== CONT  TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== CONT  TestAccAWSSagemakerNotebookInstance_kms
=== CONT  TestAccAWSSagemakerNotebookInstance_tags
=== CONT  TestAccAWSSagemakerNotebookInstance_update
=== CONT  TestAccAWSSagemakerNotebookInstance_volumesize
    resource_aws_sagemaker_notebook_instance_test.go:152: Step 1/3 error: Error running pre-apply refresh: 2020/10/14 09:16:41 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.10.0
        
        Error: Unsupported argument
        
          on config798326402/terraform_plugin_test.tf line 19:
          19:   name          = "tf-acc-test-6054892276584607903"
        
        An argument named "name" is not expected here.
        
        
        Error: Unsupported argument
        
          on config798326402/terraform_plugin_test.tf line 20:
          20:   role_arn      = aws_iam_role.test.arn
        
        An argument named "role_arn" is not expected here.
        
        
        Error: Unsupported argument
        
          on config798326402/terraform_plugin_test.tf line 21:
          21:   instance_type = "ml.t2.medium"
        
        An argument named "instance_type" is not expected here.
        
        
        Error: Unsupported argument
        
          on config798326402/terraform_plugin_test.tf line 22:
          22:   volume_size   = 8
        
        An argument named "volume_size" is not expected here.
        
--- FAIL: TestAccAWSSagemakerNotebookInstance_volumesize (2.24s)
2020/10/14 09:16:47 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/10/14 09:16:47 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
2020/10/14 09:16:47 [WARN] Truncating attribute path of 0 diagnostics for TypeSet
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
    resource_aws_sagemaker_notebook_instance_test.go:402: Step 1/3 error: Error running apply: 2020/10/14 09:16:54 [DEBUG] Using modified User-Agent: Terraform/0.12.26 HashiCorp-terraform-exec/0.10.0
        
        Error: error waiting for sagemaker notebook instance (tf-acc-test-3392626168819335841) to create: unexpected state 'Failed', wanted target 'InService'. last error: %!s(<nil>)
        
        
--- FAIL: TestAccAWSSagemakerNotebookInstance_direct_internet_access (100.82s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (360.92s)
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (380.76s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (407.11s)
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (416.13s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (696.03s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (736.92s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (1044.74s)
--- PASS: TestAccAWSSagemakerNotebookInstance_default_code_repository (1154.24s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	1154.308s
FAIL
GNUmakefile:26: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

The `TestAccAWSSagemakerNotebookInstance_direct_internet_access` failure maybe related to recent feature?

* https://aws.amazon.com/about-aws/whats-new/2020/10/now-launch-amazon-sagemaker-in-your-amazon-vpc/
* https://github.com/aws/aws-sdk-go/releases/tag/v1.35.3
* https://github.com/terraform-providers/terraform-provider-aws/pull/15500

**Comment by DrFaust92** [comment]
fixing the volume test + adding logic to recreate when volume size is lowered

**Comment by DrFaust92** [comment]
@ewbankkit, looking into error. If it related to the change then I think its only some type of validation. in the console a security group for outbound traffic/NAT is request. im not sure if its a recommendation or a requirement yet. hopefully a config change + docs is enough for this.

**Comment by DrFaust92** [comment]
Update, I don't get an issue when creating the same thing through the console. ill see what needs to be changed in the config

**Comment by DrFaust92** [comment]
added IGW for network test. passes now.

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_'==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_ -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_update
=== PAUSE TestAccAWSSagemakerNotebookInstance_update
=== RUN   TestAccAWSSagemakerNotebookInstance_volumesize
=== PAUSE TestAccAWSSagemakerNotebookInstance_volumesize
=== RUN   TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== PAUSE TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== RUN   TestAccAWSSagemakerNotebookInstance_tags
=== PAUSE TestAccAWSSagemakerNotebookInstance_tags
=== RUN   TestAccAWSSagemakerNotebookInstance_kms
=== PAUSE TestAccAWSSagemakerNotebookInstance_kms
=== RUN   TestAccAWSSagemakerNotebookInstance_disappears
=== PAUSE TestAccAWSSagemakerNotebookInstance_disappears
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== RUN   TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== RUN   TestAccAWSSagemakerNotebookInstance_default_code_repository
=== PAUSE TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
=== CONT  TestAccAWSSagemakerNotebookInstance_kms
=== CONT  TestAccAWSSagemakerNotebookInstance_tags
=== CONT  TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== CONT  TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_volumesize
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_update
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (338.80s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (370.50s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (417.09s)
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (443.71s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (746.39s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (756.86s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (776.17s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (1016.27s)
--- PASS: TestAccAWSSagemakerNotebookInstance_default_code_repository (1120.82s)
--- PASS: TestAccAWSSagemakerNotebookInstance_volumesize (1167.07s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1167.168s
```

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_' 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_ -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_update
=== PAUSE TestAccAWSSagemakerNotebookInstance_update
=== RUN   TestAccAWSSagemakerNotebookInstance_volumesize
=== PAUSE TestAccAWSSagemakerNotebookInstance_volumesize
=== RUN   TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== PAUSE TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== RUN   TestAccAWSSagemakerNotebookInstance_tags
=== PAUSE TestAccAWSSagemakerNotebookInstance_tags
=== RUN   TestAccAWSSagemakerNotebookInstance_kms
=== PAUSE TestAccAWSSagemakerNotebookInstance_kms
=== RUN   TestAccAWSSagemakerNotebookInstance_disappears
=== PAUSE TestAccAWSSagemakerNotebookInstance_disappears
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== RUN   TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== RUN   TestAccAWSSagemakerNotebookInstance_default_code_repository
=== PAUSE TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
=== CONT  TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== CONT  TestAccAWSSagemakerNotebookInstance_kms
=== CONT  TestAccAWSSagemakerNotebookInstance_tags
=== CONT  TestAccAWSSagemakerNotebookInstance_update
=== CONT  TestAccAWSSagemakerNotebookInstance_volumesize
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (346.66s)
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (360.59s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (386.54s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (397.32s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (660.18s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (744.06s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (876.80s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (932.05s)
--- PASS: TestAccAWSSagemakerNotebookInstance_volumesize (935.18s)
--- PASS: TestAccAWSSagemakerNotebookInstance_default_code_repository (1022.98s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1023.054s
```

**Comment by DrFaust92** [comment]
updated waiter:

```
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (350.17s)
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (364.47s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (463.21s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (1037.91s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (795.83s)
--- PASS: TestAccAWSSagemakerNotebookInstance_default_code_repository (1050.26s)
--- PASS: TestAccAWSSagemakerNotebookInstance_volumesize (1096.37s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (655.68s)
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (358.94s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (736.94s)
```

File: aws/resource_aws_sagemaker_notebook_instance_test.go [hunk 1]
```diff
@@ -538,283 +511,174 @@ data "aws_iam_policy_document" "assume_role" {
...
-  instance_type           = "ml.t2.medium"
-  security_groups         = [aws_security_group.test.id]
-  subnet_id               = aws_subnet.sagemaker.id
-  default_code_repository = %[2]q
+func testAccAWSSagemakerNotebookInstanceConfigVolume(rName string) string {
+	return testAccAWSSagemakerNotebookInstanceBaseConfig(rName) + fmt.Sprintf(`
+resource "aws_sagemaker_notebook_instance" "test" {
+  name          = %[1]q
+  role_arn      = aws_iam_role.test.arn
+  instance_type = "ml.t3.medium"
```
Comment (YakDriver) [inline aws/resource_aws_sagemaker_notebook_instance_test.go:636]:
```suggestion
  instance_type = "ml.t2.medium"
```

File: aws/resource_aws_sagemaker_notebook_instance_test.go [hunk 2]
```diff
@@ -538,283 +511,174 @@ data "aws_iam_policy_document" "assume_role" {
...
-      type        = "Service"
-      identifiers = ["sagemaker.amazonaws.com"]
-    }
-  }
+func testAccAWSSagemakerNotebookInstanceConfigDefaultCodeRepository(rName string, defaultCodeRepository string) string {
+	return testAccAWSSagemakerNotebookInstanceBaseConfig(rName) + fmt.Sprintf(`
+resource "aws_sagemaker_notebook_instance" "test" {
+  name                    = %[1]q
+  role_arn                = aws_iam_role.test.arn
+  instance_type           = "ml.t3.medium"
```
Comment (YakDriver) [inline aws/resource_aws_sagemaker_notebook_instance_test.go:647]:
```suggestion
  instance_type           = "ml.t2.medium"
```

File: aws/resource_aws_sagemaker_notebook_instance_test.go [hunk 3]
```diff
@@ -538,283 +511,174 @@ data "aws_iam_policy_document" "assume_role" {
...
-  vpc_id     = aws_vpc.test.id
-  cidr_block = "10.0.0.0/24"
-
-  tags = {
-    Name = "tf-acc-test-sagemaker-notebook-instance-default-code-repository"
-  }
+resource "aws_sagemaker_notebook_instance" "test" {
+  name          = %[1]q
+  role_arn      = aws_iam_role.test.arn
+  instance_type = "ml.t3.medium"
```
Comment (YakDriver) [inline aws/resource_aws_sagemaker_notebook_instance_test.go:680]:
```suggestion
  instance_type = "ml.t2.medium"
```

**Review by YakDriver — CHANGES_REQUESTED** [review]
Seems like a step backwards but capacity in commercial and GovCloud seems better.

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_ -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_update
=== PAUSE TestAccAWSSagemakerNotebookInstance_update
=== RUN   TestAccAWSSagemakerNotebookInstance_volumesize
=== PAUSE TestAccAWSSagemakerNotebookInstance_volumesize
=== RUN   TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== PAUSE TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== RUN   TestAccAWSSagemakerNotebookInstance_tags
=== PAUSE TestAccAWSSagemakerNotebookInstance_tags
=== RUN   TestAccAWSSagemakerNotebookInstance_kms
=== PAUSE TestAccAWSSagemakerNotebookInstance_kms
=== RUN   TestAccAWSSagemakerNotebookInstance_disappears
=== PAUSE TestAccAWSSagemakerNotebookInstance_disappears
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== RUN   TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== RUN   TestAccAWSSagemakerNotebookInstance_default_code_repository
=== PAUSE TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
=== CONT  TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== CONT  TestAccAWSSagemakerNotebookInstance_kms
=== CONT  TestAccAWSSagemakerNotebookInstance_tags
=== CONT  TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_volumesize
=== CONT  TestAccAWSSagemakerNotebookInstance_update
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (352.30s)
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (419.12s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (476.63s)
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (479.74s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (778.20s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (783.59s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (802.84s)
--- PASS: TestAccAWSSagemakerNotebookInstance_volumesize (945.62s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (965.70s)
--- PASS: TestAccAWSSagemakerNotebookInstance_default_code_repository (1147.09s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1149.089s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSSagemakerNotebookInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSSagemakerNotebookInstance_ -timeout 120m
=== RUN   TestAccAWSSagemakerNotebookInstance_basic
=== PAUSE TestAccAWSSagemakerNotebookInstance_basic
=== RUN   TestAccAWSSagemakerNotebookInstance_update
=== PAUSE TestAccAWSSagemakerNotebookInstance_update
=== RUN   TestAccAWSSagemakerNotebookInstance_volumesize
=== PAUSE TestAccAWSSagemakerNotebookInstance_volumesize
=== RUN   TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== PAUSE TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== RUN   TestAccAWSSagemakerNotebookInstance_tags
=== PAUSE TestAccAWSSagemakerNotebookInstance_tags
=== RUN   TestAccAWSSagemakerNotebookInstance_kms
=== PAUSE TestAccAWSSagemakerNotebookInstance_kms
=== RUN   TestAccAWSSagemakerNotebookInstance_disappears
=== PAUSE TestAccAWSSagemakerNotebookInstance_disappears
=== RUN   TestAccAWSSagemakerNotebookInstance_root_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_root_access
=== RUN   TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== PAUSE TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== RUN   TestAccAWSSagemakerNotebookInstance_default_code_repository
=== PAUSE TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_basic
=== CONT  TestAccAWSSagemakerNotebookInstance_disappears
=== CONT  TestAccAWSSagemakerNotebookInstance_kms
=== CONT  TestAccAWSSagemakerNotebookInstance_default_code_repository
=== CONT  TestAccAWSSagemakerNotebookInstance_direct_internet_access
=== CONT  TestAccAWSSagemakerNotebookInstance_root_access
=== CONT  TestAccAWSSagemakerNotebookInstance_LifecycleConfigName
=== CONT  TestAccAWSSagemakerNotebookInstance_tags
=== CONT  TestAccAWSSagemakerNotebookInstance_update
=== CONT  TestAccAWSSagemakerNotebookInstance_volumesize
--- PASS: TestAccAWSSagemakerNotebookInstance_basic (336.33s)
--- PASS: TestAccAWSSagemakerNotebookInstance_tags (340.61s)
--- PASS: TestAccAWSSagemakerNotebookInstance_disappears (349.21s)
--- PASS: TestAccAWSSagemakerNotebookInstance_kms (354.04s)
--- PASS: TestAccAWSSagemakerNotebookInstance_direct_internet_access (594.36s)
--- PASS: TestAccAWSSagemakerNotebookInstance_root_access (617.10s)
--- PASS: TestAccAWSSagemakerNotebookInstance_update (637.42s)
--- PASS: TestAccAWSSagemakerNotebookInstance_volumesize (992.65s)
--- PASS: TestAccAWSSagemakerNotebookInstance_default_code_repository (1011.40s)
--- PASS: TestAccAWSSagemakerNotebookInstance_LifecycleConfigName (1072.69s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1074.820s
```

**Pull request merged**